crates/datasources/src/object_store/mod.rs

```diff
@@ -37,10 +38,12 @@ impl FromStr for FileType {
     }
 }
 
+#[async_trait::async_trait]
 pub trait TableAccessor: Send + Sync {
     fn store(&self) -> &Arc<dyn ObjectStore>;
 
     fn object_meta(&self) -> &Arc<ObjectMeta>;
+    async fn into_table_provider(self, predicate_pushdown: bool) -> Result<Arc<dyn TableProvider>>;
```

Comment:
imo we should remove `predicate_pushdown` (can do later). I don't know when there's a time where we _don't_ want this.

Reply:
if we remove, then couldn't we just change this to `impl TryFrom`?

Reply:
Nah, we still need it be async since there's a bit of setup done in this function for some data sources (e.g. for postgres we need to make a connection and run a query to get the schema).

Reply:
I mean would could do `IntoFuture` (https://doc.rust-lang.org/std/future/trait.IntoFuture.html), but I'm not really feelin it...